### PR TITLE
Allow returning nulls for nullable boolean columns

### PR DESCRIPTION
**Pull request by eyalhakim**

### CODE REVIEW

**Comment by pleerock** [comment]
Thank you for your PR. Please add corresponding test and apply this fix to other drivers as well

**Comment by eyalhakim** [comment]
@pleerock hi, iv'e added the fix in all other drivers, but couldn't find the right testing section to implement my test in.

Could you point me at the right direction? I don't want to set up a new testbed for this, don't really have a deep knowledge of this source code.

File: src/driver/mysql/MysqlDriver.ts [hunk 1]
```diff
@@ -231,7 +231,7 @@ export class MysqlDriver implements Driver {
      */
     prepareHydratedValue(value: any, columnMetadata: ColumnMetadata): any {
         if (columnMetadata.type === Boolean) {
-            return value ? true : false;
```
Comment (pleerock) [inline src/driver/mysql/MysqlDriver.ts]: some driver (like mysql) does not have boolean type and return 1 or 0 so we have to cast it to boolean. Thats why your tests are failing

**Comment by pleerock** [comment]
You don't need to setup anything. If you have clonned repo already you simply need to create `ormconfig.json` in project root from `ormconfig.json.dist` do `docker-compose up` (if don't have install docker) and run tests. Once you get tests running simply create a new directory for this test and copy/paste any other test and an example (template) and write your tests.

**Comment by eyalhakim** [comment]
@pleerock when trying to run the docker project i get the following error:
ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint typeorm-postgres
(043345073cea4783ccc394b9e2adb66b5f0f0a29349cf3f82fd287bcbca457c1): Error starting userland proxy: Bind for 0.0.0.0:5432 failed:
port is already allocated

This is probably due to the fact that i already have postgres installed locally.
How do you suggest to get around this?

Thanks

**Comment by eyalhakim** [comment]
@pleerock although, i am seeing this in the installation console:
Creating typeorm-mongodb ...
Creating typeorm-postgres ...
Creating typeorm-mariadb ...
Creating typeorm-mysql ...
Creating typeorm-postgres
Creating typeorm-mariadb
Creating typeorm-mongodb
Creating typeorm-postgres ... error

can it be that it is trying to install postgres module twice?

**Comment by eyalhakim** [comment]
@pleerock okay, i manually stopped postgres on my machine and ran docker.
It' stuck on:

typeorm-postgres | PostgreSQL init process complete; ready for start up.
typeorm-postgres |
typeorm-postgres | LOG:  database system was shut down at 2017-07-01 09:01:29 UTC
typeorm-postgres | LOG:  MultiXact member wraparound protections are now enabled
typeorm-postgres | LOG:  database system is ready to accept connections
typeorm-postgres | LOG:  autovacuum launcher started

File: src/driver/mysql/MysqlDriver.ts (hunk 1 above)
Comment (eyalhakim) [inline src/driver/mysql/MysqlDriver.ts]: now another test is failing. However, i cannot find this test in the github-issues folder

**Comment by eyalhakim** [comment]
@pleerock hi, i'm getting the following error when running tests:
typeorm-postgres | FATAL:  password authentication failed for user "root"
typeorm-postgres | DETAIL:  Role "root" does not exist.
typeorm-postgres | 	Connection matched pg_hba.conf line 95: "host all all all md5"

**Comment by pleerock** [comment]
try to change username from `root` to `test`

**Comment by eyalhakim** [comment]
@pleerock yup, works, but now another test is failing. any idea why?

File: src/driver/mysql/MysqlDriver.ts [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
-import {ColumnMetadata} from "../../metadata/ColumnMetadata";
-import {DriverOptionNotSetError} from "../../error/DriverOptionNotSetError";
-import {DateUtils} from "../../util/DateUtils";
-import {PlatformTools} from "../../platform/PlatformTools";
-import {Connection} from "../../connection/Connection";
-import {RdbmsSchemaBuilder} from "../../schema-builder/RdbmsSchemaBuilder";
-import {MysqlConnectionOptions} from "./MysqlConnectionOptions";
-import {MappedColumnTypes} from "../types/MappedColumnTypes";
-import {ColumnType} from "../types/ColumnTypes";
+import { Driver } from "../Driver";
```
Comment (pleerock) [inline src/driver/mysql/MysqlDriver.ts]: please avoid doing any changes not needed, it makes me harder to review your code and breaks current code style.

Comment (eyalhakim) [inline src/driver/mysql/MysqlDriver.ts]: Auto formatted by my IDE. Will close this pr and open a new one without those

**Pull request closed without merging**